libvast/src/system/partition_transformer.cpp

```diff
@@ -291,41 +340,66 @@ partition_transformer_actor::behavior_type partition_transformer(
     },
     [self](atom::internal, atom::resume, atom::done, vast::id offset) {
       VAST_DEBUG("{} got new offset {}", *self, offset);
-      self->state.data.offset = offset;
-      for (auto& slice : self->state.slices) {
-        slice.offset(offset);
-        offset += slice.rows();
-        self->state.add_slice(slice);
-        self->state.stage->out().push(slice);
+      for (auto& [_, data] : self->state.data) {
+        data.offset = offset;
+        for (auto& slice : self->state.slices) {
+          slice.offset(offset);
+          offset += slice.rows();
+          self->state.add_slice(slice);
+          self->state.stage->out().push(slice);
+        }
+      }
```

Comment:
I don't understand this. Why do we push every slice for every unique layout?

Reply:
You're right, it's a bug, we should only consider the slices with the correct layout here.